cli/flox-activations/src/cli/activate/mod.rs

```diff
@@ -169,6 +226,96 @@ impl ActivateArgs {
         }
     }
 
+    fn startup_ctx(
+        ctx: ActivateCtx,
+        invocation_type: InvocationType,
+        rc_path: Option<PathBuf>,
+        env_diff: EnvDiff,
+        state_dir: &Path,
+        activate_tracer: &str,
+        subsystem_verbosity: u32,
+    ) -> Result<StartupCtx> {
+        let is_sourcing_rc = std::env::var("_flox_sourcing_rc").is_ok_and(|val| val == "true");
+        let flox_activations = ctx.interpreter_path.join("libexec/flox-activations");
+        let self_destruct =
+            !std::env::var(STARTUP_SCRIPT_NO_SELF_DESTRUCT_VAR).is_ok_and(|val| val == "true");
+
+        let clean_up = if rc_path.is_some() && self_destruct {
+            rc_path.clone()
+        } else {
+            None
+        };
+
+        let s_ctx = match ctx.shell {
+            ShellWithPath::Bash(_) => {
+                let bashrc_path = if let Some(home_dir) = dirs::home_dir() {
+                    home_dir.join(".bashrc")
+                } else {
+                    return Err(anyhow!("failed to get home directory"));
+                };
+                match invocation_type {
+                    InvocationType::InPlace => todo!(),
+                    InvocationType::Interactive | InvocationType::Command => {
+                        let startup_args = BashStartupArgs {
+                            flox_activate_tracelevel: subsystem_verbosity,
+                            activate_d: ctx.interpreter_path.join("activate.d"),
+                            flox_env: PathBuf::from(ctx.env.clone()),
+                            flox_env_cache: Some(ctx.env_cache.clone()),
+                            flox_env_project: ctx.env_project.clone(),
+                            flox_env_description: Some(ctx.env_description.clone()),
+                            is_in_place: ctx
+                                .invocation_type
+                                .is_some_and(|inv| inv == InvocationType::InPlace),
+                            bashrc_path: Some(bashrc_path),
+                            flox_sourcing_rc: is_sourcing_rc,
+                            flox_activate_tracer: activate_tracer.to_string(),
+                            flox_activations,
+                            clean_up,
+                        };
+                        StartupCtx {
+                            args: StartupArgs::Bash(startup_args),
+                            state_dir: state_dir.to_path_buf(),
+                            env_diff,
+                            rc_path,
+                            act_ctx: ctx,
+                        }
+                    },
+                }
+            },
+            ShellWithPath::Fish(_) => todo!(),
+            ShellWithPath::Tcsh(_) => todo!(),
+            ShellWithPath::Zsh(_) => todo!(),
```

Comment:
followup: do you want me to finish these out or are you going to?

Reply:
Discussed sync